tests/address_test.rs

```diff
@@ -1,267 +1,47 @@
-    match address::validate_message_address(&String::from("/[a/b")).err().expect("") {
-        OscError::BadAddress("Character range (square brackets) was started but not closed before the next address part started") => assert!(true),
-        _ => assert!(false)
-    }
-    // Square brackets contains /
-    match address::validate_message_address(&String::from("/[a/]b")).err().expect("") {
-        OscError::BadAddress("Character range (square brackets) was started but not closed before the next address part started") => assert!(true),
-        _ => assert!(false)
-    }
-    // Square brackets open but don't close before the address ends
-    match address::validate_message_address(&String::from("/[a")).err().expect("") {
-        OscError::BadAddress("Character range (square brackets) was started but not closed before the end of the address") => assert!(true),
-        _ => assert!(false)
-    }
-
-    // Curly brackets open and never close
-    match address::validate_message_address(&String::from("/{a/b")).err().expect("") {
-        OscError::BadAddress("String list (curly brackets) was started but not closed before the next address part started") => assert!(true),
-        _ => assert!(false)
-    }
-    // Curly brackets contains /
-    match address::validate_message_address(&String::from("/{a/}b")).err().expect("") {
-        OscError::BadAddress("String list (curly brackets) was started but not closed before the next address part started") => assert!(true),
-        _ => assert!(false)
-    }
-    // Curly brackets open but don't close before the address ends
-    match address::validate_message_address(&String::from("/{a"))
-        .err()
-        .expect("")
-    {
-        OscError::BadAddress(
-            "String list (curly brackets) was started but not closed before the end of the address",
-        ) => assert!(true),
-        _ => assert!(false),
-    }
-
-    // Curly brackets within square brackets
-    match address::validate_message_address(&String::from("/{a[foo]}/"))
-        .err()
-        .expect("")
-    {
-        OscError::BadAddress(
-            "Can not start a character range (square brackets) within string list (curly brackets)",
-        ) => assert!(true),
-        _ => assert!(false),
-    }
-
-    // Square brackets within curly brackets
-    match address::validate_message_address(&String::from("/[a{foo}]/"))
-        .err()
-        .expect("")
-    {
-        OscError::BadAddress(
-            "Can not start a string list (curly brackets) within character range (square brackets)",
-        ) => assert!(true),
-        _ => assert!(false),
-    }
+fn test_bad_address_pattern() {
+    let expected_err = "bad OSC address pattern: bad address pattern";
+    assert_eq!(Matcher::new("").unwrap_err().to_string(), expected_err);
+    assert_eq!(Matcher::new("/").unwrap_err().to_string(), expected_err);
+    assert_eq!(Matcher::new("//empty/parts/").unwrap_err().to_string(), expected_err); // FIX: will fail
```

Comment:
This test doesn't fail

```suggestion
    assert_eq!(Matcher::new("//empty/parts/").unwrap_err().to_string(), expected_err);
```